src/postgres_utils.py

```diff
@@ -0,0 +1,167 @@
+        DROP TABLE IF EXISTS :dataset_name CASCADE
+        """,
+        {
+            "dataset_name": AsIs(dataset_name),
+        },
+    )
+    vacuum_database()
+
+
+def get_schemas() -> list:
+    schema_names = execute_sql_select_query(
+        """
+        SELECT schema_name FROM information_schema.schemata
+        """
+    )
+    return sorted(schema_names["schema_name"].to_list())
+
+
+def get_schema_tables(table_schema: str) -> list:
+    table_names = execute_sql_select_query(
+        """
+        SELECT table_name FROM information_schema.tables WHERE table_schema = :table_schema
+        """,
+        {"table_schema": table_schema},
+    )
+
+    return sorted(table_names["table_name"].to_list())
+
+
+def get_table_columns(table_schema: str, table_name: str) -> list:
+    column_names = execute_sql_select_query(
+        """
+        SELECT column_name FROM information_schema.columns
+        WHERE table_schema = ':table_schema'
+        AND table_name   = ':table_name';
+        """,
+        {"table_schema": AsIs(table_schema), "table_name": AsIs(table_name)},
+    )
+    return sorted(column_names["column_name"])
+
+
+def get_table_row_count(table_schema: str, table_name: str) -> int:
+    row_counts = execute_sql_select_query(
+        """
+        SELECT c.reltuples::bigint AS row_count
+        FROM pg_class c
+        JOIN pg_namespace n ON n.oid = c.relnamespace
+        WHERE c.relname = ':table_name'
+        AND n.nspname = ':table_schema';
+        """,
+        {"table_schema": AsIs(table_schema), "table_name": AsIs(table_name)},
+    )
+    return int(row_counts["row_count"][0])
+
+
+def create_postigs_extension() -> None:
+    query = "CREATE EXTENSION POSTGIS"
+    execute_sql_query(query)
+
+
+def vacuum_database() -> None:
+    query = "VACUUM (ANALYZE)"
+    execute_sql_query(query)
+
+
+def create_sql_schema(table_schema: str) -> pd.DataFrame:
+    execute_sql_query(
+        "DROP SCHEMA IF EXISTS :table_schema CASCADE",
+        {"table_schema": AsIs(table_schema)},
+    )
+    execute_sql_query(
+        "CREATE SCHEMA :table_schema", {"table_schema": AsIs(table_schema)}
+    )
+    table_schema_name = execute_sql_select_query(
+        """
+        SELECT schema_name
+        FROM :table_schema.schemata;
+        """,
+        {"table_schema": AsIs("information_schema")},
+    )
+    return table_schema_name
+
+
+def execute_sql_select_query(query: str, placeholders: dict = {}) -> pd.DataFrame:
+    sql_engine = create_engine(BUILD_ENGINE)
+    with sql_engine.begin() as sql_conn:
+        select_records = pd.read_sql(sql=text(query), con=sql_conn, params=placeholders)
+    sql_engine.dispose()
+    return select_records
+
+
+def execute_sql_query(query: str, placeholders: dict = {}) -> None:
+    sql_engine = create_engine(BUILD_ENGINE, isolation_level="AUTOCOMMIT")
+    with Session(sql_engine) as session:
+        session.execute(statement=text(query), params=placeholders)
+        session.commit()
```

Comment:
agree it might as well be a connection rather than a session

but I don't think all queries return records (`CREATE TABLE`, `DROP TABLE`, `VACUUM` ?) and it seems nice being able to use `pd.read_sql` for SELECT queries to always return records as a `DataFrame` and potentially utilize the parameters that function has